salutations.php

```diff
@@ -248,36 +248,60 @@ function salutation_process_helper($contact_id, $action) {
   $core_greeting_types = array("salutation_email_greeting", "salutation_postal_greeting", "salutation_addressee");
 
   //Salutation Type Field
-  $type_field_id = civicrm_api3('CustomField', 'getvalue', [
-    'return' => "id",
-    'name' => "salutation_type",
-  ]);
+  static $type_field_id = 0;
+  if (!$type_field_id) {
+    $type_field_id = civicrm_api3('CustomField', 'getvalue', [
+      'return' => "id",
+      'name' => "salutation_type",
+    ]);
+  }
   //Salutation Types
-  $salutation_types = civicrm_api3('OptionValue', 'get',[
-    'return' => ["name","value"],
-    'option_group_id.name' => "salutation_type_options",
-    'is_active' => 1,
-  ]);
+  static $salutation_types = [];
+  if (empty($salutation_types)) {
+    $salutation_types = civicrm_api3('OptionValue', 'get',[
+      'return' => ["name","value"],
+      'option_group_id.name' => "salutation_type_options",
+      'is_active' => 1,
+    ]);
+  }
 
-  foreach($salutation_types['values'] as $salutation_type) {
```

Comment:
The biggest change is removing this `foreach`.  This makes at least one and several API calls per possible salutation type.  Since AGBU has 22 salutation types, that's 22 API calls even when nothing updates.  Later, I pull in just the salutation types in use and put those into a `foreach`.